rustls/src/common_state.rs

```diff
@@ -533,25 +533,9 @@ impl CommonState {
         &mut self,
         outgoing_tls: &mut [u8],
     ) -> Result<usize, EncryptError> {
-        debug_assert!(self.record_layer.is_encrypting());
-
-        let m = Message::build_alert(AlertLevel::Warning, AlertDescription::CloseNotify).into();
-
-        let iter = self
-            .message_fragmenter
-            .fragment_message(&m);
-
-        self.check_required_size(outgoing_tls, iter)?;
-
-        debug!("Sending warning alert {:?}", AlertDescription::CloseNotify);
-
-        let iter = self
-            .message_fragmenter
-            .fragment_message(&m);
-
-        let written = self.write_fragments(outgoing_tls, iter);
-
-        Ok(written)
+        self.send_close_notify();
```

Comment:
What was your impression of the `debug_assert!()` that is removed with the change to re-using `send_close_notify()`?

Reply:
I think this whole codepath was only reachable if `CommonState::may_send_application_data` was true, which implies `self.record_layer.is_encrypting()` is true. In any case, the code underneath `send_close_notify` works for both cases (encrypted alerts, and not)